### PR TITLE
[geo] Add Deckgl GeoJson layer

### PR DESCRIPTION
**Pull request by hughhhh**
Added DeckGLs GeoJson layer as a new visualization type

@mistercrunch

### CODE REVIEW

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 1]
```diff
@@ -514,6 +514,16 @@ export const controls = {
     }),
   },
 
+  geojson: {
+    type: 'SelectControl',
+    label: t('GeoJSON'),
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: "GeoJSON column"

File: superset/assets/javascripts/explore/stores/visTypes.js [hunk 2]
```diff
@@ -431,6 +431,40 @@ export const visTypes = {
...
+        ],
+      },
+      {
+        label: t('Map'),
+        controlSetRows: [
+          ['mapbox_style', 'viewport'],
+        ],
+      },
+      {
+        label: t('Grid'),
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/visTypes.js]: This section/controls are not relevant for the GeoJSON viz

File: superset/assets/javascripts/explore/stores/visTypes.js [hunk 3]
```diff
@@ -431,6 +431,40 @@ export const visTypes = {
...
+        controlSetRows: [
+          ['grid_size', 'color_picker'],
+        ],
+      },
+    ],
+    controlOverrides: {
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/visTypes.js]: This override is not relevant here

File: superset/assets/visualizations/deckgl/geojson.jsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import DeckGLContainer from './DeckGLContainer';
+
+function DeckGeoJsonLayer(slice, payload, setControlValue) {
+  const fd = slice.formData;
+  const c = fd.color_picker;
+  const data = payload.data.geojson.features.map(d => ({
+      ...d,
+      properties: {
+        fillColor: [c.r, c.g, c.b, 255 * c.a],
```
Comment (mistercrunch) [inline superset/assets/visualizations/deckgl/geojson.jsx]: The GeoJSON may have its own colors defined, if that's the case the user may or may not want to override it. I feel like we'll need an intricate "GeoJSON properties" control where the user could decide to override or not properties like fillColor, strokeColor, strokeWidth, ...

https://github.com/uber/deck.gl/blob/master/docs/layers/geojson-layer.md

**Review by mistercrunch — CHANGES_REQUESTED** [review]
We'll have to do a bit of thinking as to how we allow people to define/override the properties of the GeoJSON

File: superset/assets/visualizations/main.js [hunk 5]
```diff
@@ -42,49 +42,51 @@ export const VIZ_TYPES = {
...
-  [VIZ_TYPES.world_map]: require('./world_map.js'),
-  [VIZ_TYPES.dual_line]: require('./nvd3_vis.js'),
-  [VIZ_TYPES.event_flow]: require('./EventFlow.jsx'),
-  [VIZ_TYPES.paired_ttest]: require('./paired_ttest.jsx'),
-  [VIZ_TYPES.partition]: require('./partition.js'),
-  [VIZ_TYPES.deck_scatter]: require('./deckgl/scatter.jsx'),
-  [VIZ_TYPES.deck_screengrid]: require('./deckgl/screengrid.jsx'),
-  [VIZ_TYPES.deck_grid]: require('./deckgl/grid.jsx'),
-  [VIZ_TYPES.deck_hex]: require('./deckgl/hex.jsx'),
+  area: require('./nvd3_vis.js'),
```
Comment (mistercrunch) [inline superset/assets/visualizations/main.js]: Let's use `VIZ_TYPES` here, I think you picked the wrong side of the merge conflict.

File: superset/assets/visualizations/deckgl/geojson.jsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+import DeckGLContainer from './DeckGLContainer';
+
+function DeckGeoJsonLayer(slice, payload, setControlValue) {
+  const fd = slice.formData;
+  const c = fd.color_picker;
+  const data = payload.data.geojson.features.map(d => ({
+    ...d,
+    properties: {
+      fillColor: [c.r, c.g, c.b, 255 * c.a],
```
Comment (mistercrunch) [inline superset/assets/visualizations/deckgl/geojson.jsx]: I still feel like there should be a way to not override the color defined in the GeoJSON (if any). We need better mapping and precedence rules here.

First, a mapping from the spec to deck.gl's supported props.

Then, clearable colorpickers (we could use opacity=0 as a hint to not override, opacity=0 could be the default), only if opacity!=0 do we override the GeoJSON provided colors.

As a last resort, we should have a color if opacity=0 and color isn't defined in the geojson.

All of this should be made clear in the controls tooltip.

This is hard to model, let's chat about it.

**Comment by hughhhh** [comment]
Moving -> #4097

**Pull request closed without merging**